universal-login-react/src/core/services/WalletCreationService.ts

```diff
@@ -1,27 +1,31 @@
 import {WalletService} from '@universal-login/sdk';
-import {GasParameters, INITIAL_GAS_PARAMETERS} from '@universal-login/commons';
+import {ensure, GasParameters, INITIAL_GAS_PARAMETERS} from '@universal-login/commons';
+import {waitFor} from 'reactive-properties';
 
 export class WalletCreationService {
   private gasParameters: GasParameters = INITIAL_GAS_PARAMETERS;
-  private callbackOnBalancePresent: (() => void) | undefined;
 
   constructor(
     private readonly walletService: WalletService,
   ) {
   }
 
-  async initiateCreationFlow(ensName: string) {
-    const {waitForBalance} = await this.walletService.createFutureWallet(ensName);
-    await waitForBalance();
-    this.callbackOnBalancePresent && this.callbackOnBalancePresent();
+  async deployWhenReady(onBalancePresent?: () => void) {
+    const state = await this.walletService.stateProperty.pipe(
+      waitFor(state => state.kind === 'Future'),
```

Comment:
Idea:
`state.isType('Future')`

Reply:
This would require us to turn the state into a class. I think we need to further consider the benefits and implications that would give us.